docs/initial-design.md

```diff
@@ -0,0 +1,72 @@
+# Proposal: Design of External DNS
+
+## Background
+
+[Project proposal](https://groups.google.com/forum/#!searchin/kubernetes-dev/external$20dns%7Csort:relevance/kubernetes-dev/2wGQUB0fUuE/9OXz01i2BgAJ)
+
+[Initial discussion](https://docs.google.com/document/d/1ML_q3OppUtQKXan6Q42xIq2jelSoIivuXI8zExbc6ec/edit#heading=h.1pgkuagjhm4p)
+
+This document describes the initial design proposal
+
+External DNS is purposed to fill the existing gap of creating DNS records for Kubernetes resources. While there exist alternative solutions, this project is meant to be a standard way of managing DNS records for Kubernetes. The current project is a fusion of the following projects and driven by its maintainers:
+
+1. [Kops DNS Controller](https://github.com/kubernetes/kops/tree/master/dns-controller)
+2. [Mate](https://github.com/zalando-incubator/mate)
+3. [wearemolecule/route53-kubernetes](https://github.com/wearemolecule/route53-kubernetes)
+
+## Example use case:
+
+User runs `kubectl create -f ingress.yaml`, this will create an ingress as normal.
+Typically the user would then have to manually create a DNS record pointing the ingress endpoint
+If the external-dns controller is running on the cluster, it could automatically configure the DNS records instead, by observing the host attribute in the ingress object.
```

Comment:
@iterion yup, seems good to me. You can push to this branch, this PR is for us to agree on same design basis :) we can fix it any time later